### PR TITLE
Purchases: Update TransactionsHeader to use Redux analytics and React.Component

### PR DESCRIPTION
**Pull request by tyxla**
This PR refactors `TransactionsHeader` to:

* Use Redux analytics instead of `eventRecorder`.
* Remove all mixins, specifically `eventRecorder`.
* Convert from `createReactClass` to a `React.Component`.
* Sort some imports.
* Update `var` usages to `let` or `const`.

This PR should not introduce any visual changes.

Part of #20053.

To test:
* Checkout this branch
* Go to http://calypso.localhost:3000/me/purchases/billing
* Verify you can observe the right analytics actions being dispatched when:
  * The "Date" popover is expanded/collapsed
  * The "All Apps" popover is expanded/collapsed
  * Any item from the "Date" popover is selected
  * Any item from the "All Apps" popover is selected
* Verify everything else on the purchases list page works like it did before.

### CODE REVIEW

File: client/me/billing-history/transactions-header.jsx [hunk 1]
```diff
@@ -249,14 +259,14 @@ const TransactionsHeader = createReactClass( {
...
+	getApps() {
 		return uniq( map( this.props.transactions, 'service' ) );
-	},
+	}
 
-	renderAppPicker: function( title, app, analyticsEvent ) {
-		var filter = { app: app },
+	renderAppPicker( title, app, analyticsEvent ) {
+		const filter = { app: app },
 			classes = classNames( {
```
Comment (sirreal) [inline client/me/billing-history/transactions-header.jsx]: Want to do independent declarations as long as you're here?

File: client/me/billing-history/transactions-header.jsx [hunk 2]
```diff
@@ -185,34 +201,31 @@ const TransactionsHeader = createReactClass( {
...
 
 		return tableRows.filter( this.props.transactions, filter ).length;
-	},
+	}
 
-	renderAppsPopover: function() {
-		var isVisible = 'apps' === this.state.activePopover,
+	renderAppsPopover() {
+		const isVisible = 'apps' === this.state.activePopover,
 			classes = classNames( {
```
Comment (sirreal) [inline client/me/billing-history/transactions-header.jsx]: Split declarations?

Comment (tyxla) [reply client/me/billing-history/transactions-header.jsx]: 👍  d252e6781df12401f40ca9ccd9288b88adaaefba

File: client/me/billing-history/transactions-header.jsx (hunk 1 above)
Comment (tyxla) [reply client/me/billing-history/transactions-header.jsx]: 👍  d252e6781df12401f40ca9ccd9288b88adaaefba

**Comment by tyxla** [comment]
Thanks @sirreal!

**Pull request merged**